### ISSUE TITLE
How to apply margins between pages?

### ISSUE THREAD

**Issue by RadomirNowak** [react-pdf-viewer/react-pdf-viewer#1157]
Hi there!
I was wondering about the proper way to apply a margin between pages. My approach was to add margin to `rpv-core__inner-page` class, but that does not seem to work (page navigation gets broken). At the moment, when I render my PDF i get very little space between pages (see screenshot)
![image](https://user-images.githubusercontent.com/11005257/172594808-fd5cea71-77e3-47d1-bf91-49ab8a7c3197.png)

Thanks in advance!

**Comment by phuoc-ng** [comment]
It's not possible with the latest version. Making no spaces between pages will ensure the scrolling/zooming work perfectly.
You can try to override the following styles

```css
.rpv-core__page-layer::after {
    bottom: 0.5rem;
    left: 0.5rem;
    right: 0.5rem;
    top: 0.5rem;
}
```

**Comment by RadomirNowak** [comment]
Thanks @phuoc-ng for explaining :bow: out of curiosity - are there any plans to support having some spacing between pages? I tried the overriding example that you suggested but unfortunately it did not work

**Comment by phuoc-ng** [comment]
I created this [sample](https://codesandbox.io/s/react-pdf-viewer-page-spacing-ldic1l?file=/src/App.tsx) project.
Notice that the custom styles must be placed after the React PDF Viewer styles

```css
import "@react-pdf-viewer/core/lib/styles/index.css";
import "@react-pdf-viewer/default-layout/lib/styles/index.css";

// The styles that overrides React PDF Viewer styles
// Must be placed at the bottom
import "./styles.css";
```

![Screen Shot 2022-06-14 at 17 30 14](https://user-images.githubusercontent.com/57786711/173557240-6f4e0de6-f0cb-4deb-9877-c07e1e6f5b9c.png)

**Comment by ankit2049** [comment]
@phuocng we faced a similar issue after upgrading to the latest version. We are working around by overriding some css styles but it's creating rendering issues in some of the files. Is there a plan to support this anytime in future?

**Comment by phuocng** [comment]
@ankit2049 @RadomirNowak @baptistefkt

1. The request was implemented in the latest code and will be available in the next v3.9.0.

https://user-images.githubusercontent.com/57786711/202892073-e2594625-4cf8-4e1e-b140-dc18649e3ede.mp4

2. API:

```jsx
const pageLayout: PageLayout = {
    tranformSize: ({ size }) => ({ height: size.height + 30, width: size.width + 30 }),
};

<Viewer pageLayout={pageLayout} />
```

There is no single option such as `padding`, because `PageLayout` will provide more useful options later.

It ensures all rendering and navigating between pages/annotations/bookmarks work properly.

**Comment by phuocng** [comment]
Done in [v3.9.0](https://react-pdf-viewer.dev/updates/). I also created this [example](https://react-pdf-viewer.dev/examples/add-margin-between-pages/) for you.
